package.json

```diff
@@ -36,7 +36,8 @@
     "prepublish": ". ./resources/prepublish.sh"
   },
   "dependencies": {
-    "iterall": "1.0.2"
+    "iterall": "1.0.2",
+    "moment": "^2.15.2"
```

Comment:
I completely agree with you on this one! I had some doubts about whether to include `moment` as a dependency because it is only used for a very small proportion of the code. There are some features of `moment` that I used:

- Checking for a valid date is very easy using the `isValid()` function. I ran into the problem that the invalid date `2015-02-29` was regarded as a valid date in the `Date` object (because days are numbered from 1-31).
- Checking for correctly formatted date strings is very easy in `moment` (you can specify an array of valid date formats). I initially tried using a single validating regex to validate all the supported date formats but it quickly became too complex to understand. 

I will try to overcome these problems:) I will try to write multiple easy to understand validating regex for verifying that the date string being parsed is in ISO date format and remove the `moment` dependency.

I think that the important thing when doing these operations that the code remains readable and understandable. So I will make sure to write code this way:)